app.py

```diff
@@ -1,23 +1,42 @@
 import json
 from flask import Flask, request, jsonify
 from flask.views import MethodView
-from werkzeug.routing import ValidationError
+from werkzeug.exceptions import HTTPException
+from error_handling import CustomError, ValidationError, WrongIdError
 
 app = Flask(__name__)
+pattern = {'id': int, 'name': str, 'address': str, "work_time": str, "phone_number": str}
 with open("restaurants.json", "r") as read_file:
     data = json.load(read_file)
 
 restaurants = data['restaurants']
 
 
+def validate_post(content):
+    if content is None:
+        raise ValidationError(description="Required data not available(No data)")
+    if content.keys() != pattern.keys():
+        raise ValidationError(description="Sent data not correct, doesn't match format")
+    for key, value in pattern.items():
+        if type(content[key]) != value:
+            raise ValidationError(description="Type of sent data not correct")
+
+
+def validate_put(content):
+    if content is None:
+        raise ValidationError(description="Required data not available(No data)")
+    for key in content.keys():
+        if key not in pattern.keys():
+            raise ValidationError(description="Required data not correct, doesn't match format")
```

Comment:
and what if `content` has all the required fields but with wrong type? e.g. `name` is `int`

Reply:
also, compare `validate_put` and `validate_post`, their code looks quite similar. Probably (I am not sure) there is a way to extract common code parts, or even replace them both with one function in a neat way.

Reply:
fixed